contents/docs/libraries/ios/index.mdx

```diff
@@ -140,35 +140,158 @@ This will remove the Super Property and subsequent events will not include it.
+`BeforeSendBlock` gives you one place to edit or redact information before it is sent to PostHog. For example:
+
+<details>
+<summary>Redact URLs in event properties</summary>
+
+```swift
+let config = PostHogConfig(apiKey: "<ph_project_api_key>", host: "<ph_instance_address>")
+
+config.setBeforeSend { event in
+    // Redact URLs
+    if let url = event.properties["url"] as? String {
+        event.properties["url"] = url.map { _ in "*" }.joined()
+    }
+    return event
+}
+```
+</details>
+
+<details>
+<summary>Redact sensitive information from event properties</summary>
 
 ```swift
-class ExampleSanitizer: PostHogPropertiesSanitizer {
-    public func sanitize(_ properties: [String: Any]) -> [String: Any] {
-        // Create a mutable copy
-        var sanitizedProperties = properties
-        // Perform sanitization
-        // For example, removing keys with empty values
-        for (key, value) in properties {
-            if let stringValue = value as? String, stringValue.isEmpty {
-                sanitizedProperties.removeValue(forKey: key)
-            }
+let config = PostHogConfig(apiKey: "<ph_project_api_key>", host: "<ph_instance_address>")
+
+config.setBeforeSend { event in
+    // Redact sensitive information
+    if let email = event.properties["email"] as? String {
+        event.properties["email"] = email.map { _ in "*" }.joined()
+    }
+    
+    return event
+}
+
+PostHogSDK.shared.setup(config)
+```
+
+</details>
+
+<details>
+<summary>Drop events by event name</summary>
+
+```swift
+let config = PostHogConfig(apiKey: "<ph_project_api_key>", host: "<ph_instance_address>")
+
+config.setBeforeSend { event in
+    // Drop all events named "Stale Event"
+    if event.event == "Stale Event" {
+        return nil
+    }
+    return event
+}
+```
+
+</details>
+
+### Sampling events
+
+Sampling lets you choose to send only a percentage of events to PostHog. It is a good way to control your costs without having to completely turn off features of the SDK.
+
+<details>
+<summary>Sample events by event name</summary>
+
+```swift
+let config = PostHogConfig(apiKey: "<ph_project_api_key>", host: "<ph_instance_address>")
+
+config.setBeforeSend { event in
+    // Sample 10% of Sampled Event events
+    if event.event == "Sampled Event" {
+        if Double.random(in: 0...1) < 0.1 {
+            event.properties["$sample_type"] = ["sampleByEvent"]
+            event.properties["$sample_threshold"] = 0.1
+            event.properties["$sample_events"] = ["Sampled Event"]
+            return event
         }
-        return sanitizedProperties
+        return nil
     }
+    return event
 }
+
+PostHogSDK.shared.setup(config)
 ```
 
+</details>
+
+<details>
+<summary>Sample events by distinct ID</summary>
+
+```swift
+let config = PostHogConfig(apiKey: "<ph_project_api_key>", host: "<ph_instance_address>")
+
+config.setBeforeSend { event in
+    // Sample 40% of users by their distinct ID
```

Comment:
not sure if that makes sense in mobile since its always the same distinct id?

Reply:
Well each language might implement the hashValue differently so i am not sure this works correctly unless its swift across the whole stack running on the same version

Reply:
that can be, since only JS has sampling and we are adding now, if the goal is to sample per distinct id across the stack, I suggest we port this simple hash then